radixdlt-java/src/main/java/com/radixdlt/client/core/ledger/InMemoryAtomStore.java

```diff
@@ -42,26 +66,36 @@ public void store(RadixAddress address, AtomObservation atomObservation) {
 	@Override
 	public Observable<AtomObservation> getAtoms(RadixAddress address) {
 		Objects.requireNonNull(address);
-		return Observable.fromCallable(HashMap<RadixHash, Type>::new)
-			.flatMap(curAtomState ->
+		return Observable.fromCallable(AtomObservationsState::new)
+			.flatMap(atomsObservationState ->
 				cache.computeIfAbsent(address, addr -> ReplaySubject.create())
 					.filter(observation -> {
+						atomsObservationState.curCount++;
+
 						if (observation.getAtom() != null) {
-							Type curState = curAtomState.get(observation.getAtom().getHash());
-							final boolean shouldUpdate;
-							if (curState == null) {
-								shouldUpdate = observation.getType() == Type.STORE;
+							AtomObservationUpdateType nextUpdate = observation.getUpdateType();
+							AtomObservationUpdateType lastUpdate = atomsObservationState.get(observation.getAtom());
+
+							final boolean filter;
```

Comment:
I think the sense of this boolean is back to front.  It should be called `include` or something similar.